### ISSUE TITLE
No static method '_fromJson' declared in class 'LocationResult'

### ISSUE THREAD

**Issue by gorjan-mishevski** [loup-v/geolocation#3]
I was testing this plugin. I ran this code
```
import 'package:geolocation/geolocation.dart';

find() async {
    LocationResult result = await Geolocation.currentLocation(LocationAccuracy.best);
}
```

I got this error
```
[VERBOSE-2:dart_error.cc(16)] Unhandled exception:
NoSuchMethodError: No static method '_fromJson' declared in class 'LocationResult'.
Receiver: LocationResult
Tried calling: _fromJson(_LinkedHashMap len:2)
#0      NoSuchMethodError._throwNew (dart:core-patch/dart:core/errors_patch.dart:192)
#1      _Codec.decodeLocation (package:geolocation/channel/codec.dart:8:22)
#2      _Api.currentLocation (package:geolocation/channel/api.dart:20:19)
<asynchronous suspension>
#3      Geolocation.currentLocation (package:geolocation/geolocation.dart:52:12)
<asynchronous suspension>
#4      UserLocation.find (/Users/gorjan/Projects/aero/lib/models/UserLocation.dart:18:47)
<asynchronous suspension>
#5      HomeState.initState (/Users/gorjan/Projects/aero/lib/pages/home.dart:26:26)
#6      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3734:58)
#7      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3600:5)
#8      Element.inflateWidget (package:flutter/src/widgets/framework
```

Testing on iPhone X iOS 11.3

Pubspec
```
 geolocation: "0.1.1"
```

**Comment by lukaspili** [comment]
Hello @gorjan-mishevski 
I failed to reproduce your error (starting from a new project and just adding `geolocation` pub).
Can you provide a sample project where I can reproduce this issue?

**Comment by daniel-v** [comment]
This issue is related to Dart2.0 optional `new` for calling constructors. For many of the devs - including me - it is less optional for now. Using the same `beta` version of the framework.

Putting a couple of `new`s should fix it. 

Also: 
`GeolocationException`, `LocationResult`, `GeolocationResultError` are instantiated without new. Just that I found quickly.

**Comment by lukaspili** [comment]
Thanks for the catch @daniel-v, will fix this for tomorrow's release.

**Comment by lukaspili** [comment]
Fixed in 0.2.1.
